### PR TITLE
telemetry: fix err check

### PR DESCRIPTION
**Pull request by Tjianke**
### What problem does this PR solve?

Issue Number: #22979  

Problem Summary:

### What is changed and how it works?

What's Changed: added err check

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

- No release note

### CODE REVIEW

File: telemetry/util.go [hunk 1]
```diff
@@ -24,7 +24,10 @@ import (
 // hashString returns the SHA1 checksum in hex of the string.
 func hashString(text string) string {
 	hash := sha1.New()
-	hash.Write([]byte(text))
+	_, err := hash.Write([]byte(text))
```
Comment (zimulala) [inline telemetry/util.go:27]: I think we can add a return value to this function.

Comment (Tjianke) [reply telemetry/util.go:27]: Return val added and fixed relevant usages a bit.

**Review by djshow832 — APPROVED** [review]

**Pull request merged**